Build failure: iqtree

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Build iqtree

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/314852356

### Relevant log output

```console
> cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/share/doc/iqtree -DCMAKE_INSTALL_INFODIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/include -DCMAKE_INSTALL_SBINDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6/lib -DCMAKE_STRIP=/nix/store/ag7fc58qzff4j1sgd7v98irhcr6vcg8v-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/ag7fc58qzff4j1sgd7v98irhcr6vcg8v-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/ag7fc58qzff4j1sgd7v98irhcr6vcg8v-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/y5rxp232ky6j16x530ssam8y5w917pb6-iqtree-2.3.6
       > CMake Error at CMakeLists.txt:63 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       > 
       > -- Configuring incomplete, errors occurred!
```

### Additional context

It might be an consequence of an update to CMake

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.58, NixOS, 25.11 (Xantusia), 25.11.20251115.1d4c883`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/k5s8fqk94zzga84iym6wvaqnhmq7ykwf-source`

### Notify maintainers

@bzizou 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/iq/iqtree/package.nix
+++ b/pkgs/by-name/iq/iqtree/package.nix
@@ -9,15 +9,15 @@
   llvmPackages,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "iqtree";
-  version = "2.3.6";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "iqtree";
-    repo = "iqtree2";
-    tag = "v${version}";
-    hash = "sha256-8d5zqZIevv3bnq7z7Iyo/x8i445y1RAFtRMeK8s/ieQ=";
+    repo = "iqtree3";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-6Zy/u7YSCoqkLwq6xOoFMFDInGYREea1gS+PzSP4F8Q=";
     fetchSubmodules = true;
   };
 
@@ -32,15 +32,16 @@ stdenv.mkDerivation rec {
     llvmPackages.openmp
   ];
 
+  cmakeFlags = [
+    (lib.cmakeBool "USE_CMAPLE" false) # tries to download googletest
+  ];
+
   meta = {
-    homepage = "http://www.iqtree.org/";
+    homepage = "https://iqtree.github.io/";
     description = "Efficient and versatile phylogenomic software by maximum likelihood";
-    mainProgram = "iqtree2";
+    mainProgram = "iqtree3";
     license = lib.licenses.lgpl2;
     maintainers = with lib.maintainers; [ bzizou ];
-    platforms = [
-      "x86_64-linux"
-      "x86_64-darwin"
-    ];
+    platforms = lib.platforms.linux;
   };
-}
+})

